### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by mcottontensor**
## Relevant components:
- [ ] Signalling server
- [ ] Frontend library
- [ ] Frontend UI library
- [ ] Matchmaker
- [ ] Platform scripts
- [ ] SFU

## Problem statement:
Missing documentation for several components.

## Solution
Added some brief documentation that was missing.

## Documentation
Added SignallingWebServer docs. Linked the existing Matchmaker docs. Added SFU docs.

## Test Plan and Compatibility
No testing needed.

### CODE REVIEW

**Comment by mcottontensor** [comment]
Closes #257
Closes #256
Closes #255

File: SFU/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+| worker | Object | | Worker related configuration options. See below. |
+| router | Object | | Router related configuration options. See below. |
+| webRtcTransport | Object | | WebRTC transport related configuration options. See below. |
+
+### Worker related configuration options.
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| rtcMinPort | Number | 40000 | Minimun RTC port for ICE, DTLS, RTP, etc. |
+| rtcMazPort | Number | 49999 | Maximum RTC port for ICE, DTLS, RTP, etc. |
```
Comment (lukehb) [inline SFU/README.md]: Should be `rtcMaxPort`

File: SFU/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+The SFU (Selective Forwarding Unit) is a mechanism to allow distributing a single stream out to a large number of peers. This is useful because when peers connect directly to the streamer, resources must be allocated per peer to allow encoding of the stream. This means resources can quickly be drained after only a handful of peers. The SFU can receive multiple streams using simulcast and selectively forward out streams to remote peers based on their available resources, without requiring to actually reencode the stream.
+
+## Configuration
+
+Configuration is handled through the single config.js file.
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| signallingURL | String | 'http://localhost:8889' | The URL pointing to the signalling server we want to connect to. |
+| SFUId | String | 'SFU' | The streamer name this peer will be given when connecting to the signalling server. Peers wishing to receive from this SFU should subscribe to this ID. |
```
Comment (lukehb) [inline SFU/README.md]: I think 

> The ~~streamer~~ name this peer will be given when connecting to the signalling server. Peers wishing to receive from this SFU should subscribe to this ID.

File: SFU/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Several scripts are supplied for Windows and Linux in the [platform_scripts](platform_scripts/) folder. These are the easiest way to get the server running under common situations. They can also be used as a reference for new situations.
+
+## Streaming from UE
+
+The best way to fully utilize the SFU is to have a single streamer streaming simulcast to the SFU and then have peers subscribe to the SFU stream.
+
+Launch the streaming app with the following arguments
+`-SimulcastParameters="1.0,5000000,20000000,2.0,1000000,5000000,4.0,50000,1000000"`
+This tells the pixel streaming plugin to stream simulcast with 3 streams, each one scaling down by 2. The sequence of values is as follows, `scale_down_factor,min_bitrate,max_bitrate,...repeating for each stream`
```
Comment (lukehb) [inline SFU/README.md]: I prefer:

> scaling video resolution by half.

Comment (lukehb) [reply SFU/README.md]: > pixel streaming

to 

> Pixel Streaming

**Review by lukehb — CHANGES_REQUESTED** [review]
Minor changes requested.

File: SFU/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Pixel Streaming Selective Forwarding Unit
+
+The SFU (Selective Forwarding Unit) is a mechanism to allow distributing a single stream out to a large number of peers. This is useful because when peers connect directly to the streamer, resources must be allocated per peer to allow encoding of the stream. This means resources can quickly be drained after only a handful of peers. The SFU can receive multiple streams using simulcast and selectively forward out streams to remote peers based on their available resources, without requiring to actually reencode the stream.
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > This means resources can quickly be drained

This means the resources can be quickly drained

> reencode the stream

re-encode the stream

File: SFU/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+## Configuration
+
+Configuration is handled through the single config.js file.
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| signallingURL | String | 'http://localhost:8889' | The URL pointing to the signalling server we want to connect to. |
+| SFUId | String | 'SFU' | The streamer name this peer will be given when connecting to the signalling server. Peers wishing to receive from this SFU should subscribe to this ID. |
+| subscribeStreamerId | String | 'DefaultStreamer' | This is the name of the streamer that this SFU should subscribe to and restream. |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > subscribe to and restream

subscribe to and re-stream

File: SFU/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+| retrySubscribeDelaySecs | Number | 10 | If subscribing to the given streamer fails, wait this many seconds before trying again. |
+| mediasoup | Object | | Mediasoup related configuration options. See below. |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > Mediasoup related

Mediasoup-related

File: SFU/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+### Mediasoup related configuration options.
```
Comment (DenisTensorWorks) [inline SFU/README.md:17]: > Mediasoup related configuration options.

Mediasoup-related configuration options

File: SFU/README.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| worker | Object | | Worker related configuration options. See below. |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > Worker related 

Worker-related

File: SFU/README.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+| router | Object | | Router related configuration options. See below. |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > Router related 

Router-related

File: SFU/README.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+| webRtcTransport | Object | | WebRTC transport related configuration options. See below. |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > transport related

transport-related

File: SFU/README.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+### Worker related configuration options.
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > Worker related configuration options.

Worker-related configuration options

File: SFU/README.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| rtcMinPort | Number | 40000 | Minimun RTC port for ICE, DTLS, RTP, etc. |
+| rtcMazPort | Number | 49999 | Maximum RTC port for ICE, DTLS, RTP, etc. |
+| logLevel | String | 'debug' | The log level for the worker. See Mediasoup [docs](https://mediasoup.org/documentation/v3/mediasoup/api/#WorkerLogLevel) |
+| logTags | Array&lt;WorkerLogTag&gt; | | The log tags to include in logs. See Mediasoup [docs](https://mediasoup.org/documentation/v3/mediasoup/api/#WorkerLogTag) |
+
+### Router related configuration options.
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > Router related configuration options.

Router-related configuration options

File: SFU/README.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| mediaCodecs | Array&lt;RtpCodecCapability&gt; | | Codecs to support. See Mediasoup [docs](https://mediasoup.org/documentation/v3/mediasoup/rtp-parameters-and-capabilities/#RtpCodecCapability) |
+
+### WebRTC transport related configuration options.
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > WebRTC transport related configuration options.

WebRTC transport-related configuration options

File: SFU/README.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+| Name | Type | Default | Description |
+|-|-|-|-|
+| listenIps | Array&lt;TransportListenIp|String&gt; | | Listening IP address or addresses in order of preference (first one is the preferred one). See Mediasoup [docs](https://mediasoup.org/documentation/v3/mediasoup/api/#TransportListenIp) |
+| initialAvailableOutgoingBitrate | Number | Initial available outgoing bitrate (in bps). |
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > (in bps).

(in bps/bits per second)

File: SFU/README.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+## Streaming from UE
+
+The best way to fully utilize the SFU is to have a single streamer streaming simulcast to the SFU and then have peers subscribe to the SFU stream.
+
+Launch the streaming app with the following arguments
+`-SimulcastParameters="1.0,5000000,20000000,2.0,1000000,5000000,4.0,50000,1000000"`
+This tells the pixel streaming plugin to stream simulcast with 3 streams, each one scaling down by 2. The sequence of values is as follows, `scale_down_factor,min_bitrate,max_bitrate,...repeating for each stream`
+
+When this streams to the SFU the SFU will detect these 3 streams and then selectively stream these out to connected peers based on their connection quality.
```
Comment (DenisTensorWorks) [inline SFU/README.md]: > When this streams to the SFU the SFU will detect

When this streams to the SFU, the SFU will detect

File: SignallingWebServer/README.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
+# Pixel Streaming Signalling Server
+
+The signalling server is a small middle-man application that sits between streamers and other peers. It handles the initial connection negotiations and some other small ongoing control messages between peers as well as acting as a simple web server for serving the [Frontend](/Frontend/README.md) web application.
```
Comment (DenisTensorWorks) [inline SignallingWebServer/README.md]: > a small middle-man application

a small intermediary application

File: SignallingWebServer/README.md [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+
+## Configuration
+
+Configuration of the signalling server is handled via the config.js file in the SignallingWebServer directory. The following are supported options.
```
Comment (DenisTensorWorks) [inline SignallingWebServer/README.md]: > The following are supported options.

The following are its supported options.

File: SignallingWebServer/README.md [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+| UseMatchmaker | Boolean | false | Enables or disables the use of the [Matchmaker](/Matchmaker) application. |
+| UseHTTPS | Boolean | false | Enables or disables ssl for the serving of the Frontend. |
+| HTTPSCertFile | String | './certificates/client-cert.pem' | The path to the SSL cert file for when HTTPS is enabled. |
+| HTTPSKeyFile | String | './certificates/client-key.pem' | The path to the SSL key file for when HTTPS is enabled. |
+| LogToFile | Boolean | true | Enable or disable logging to a file in the 'logs' folder. |
+| LogVerbose | Boolean | true | Enable or disable verbose logging. Adds a lot of extra information to logs. |
+| HomepageFile | String | 'player.html' | The root file of the frontend web application. |
+| AdditionalRoutes | Map | | Additional routes for the web application. |
+| EnableWebserver | Boolean | true | Enables or disables the serving of the frontend through the internal web server. Disbable this if you are serving your own frontend. |
+| MatchmakerAddress | String | | The IP/hostname of the match maker application. |
```
Comment (DenisTensorWorks) [inline SignallingWebServer/README.md]: > the match maker application

the matchmaker application

File: SignallingWebServer/README.md [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+| MatchmakerPort | Number | 9999 | The port the match maker is listening on. |
```
Comment (DenisTensorWorks) [inline SignallingWebServer/README.md]: > the match maker is listening on

the matchmaker is listening on

File: SignallingWebServer/README.md [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+| PublicIp | String | "localhost" | The public IP/hostname of the host that the signalling server is listening on. This is used by the match maker. |
```
Comment (DenisTensorWorks) [inline SignallingWebServer/README.md]: > This is used by the match maker.

This is used by the matchmaker.

**Review by DenisTensorWorks — COMMENTED** [review]
Minor style and sentence flow edits.

**Pull request merged**